File: MBStore/Models/CartItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MBStore.Models
{
    public class CartItem
    {
        public Guid CartItemId { get; set; }

        public IList<CartItemProduct> CartItemProducts { get; set; }

        public int Quantity { get; set; }

        public decimal Subtotal { get; set; }

        public Guid CartId { get; set; }

        public Cart Cart { get; set; }

        public CartItem()
        {
            CartItemId = Guid.NewGuid();
        }

    }
}

File: MBStore/ClientApp/src/app/shared/CartItem.ts
import { CartItemProduct } from './CartItemProduct';
import { Cart } from './Cart';

export class CartItem {
    public cartItemId: string;
    public cartItemProducts: CartItemProduct[];
    public quantity: number;
    public subTotal: number;
    public cartId: string;
    public cart: Cart;
}